filefetcher/fetch.py

```diff
@@ -21,46 +27,144 @@
+        if dsname not in fdict:
+            fdict[dsname] = [args.xrd + f for f in flist if len(f) > 1]
+        else:  # needed to collect all data samples into one common key "Data" (using append() would introduce a new element for the key)
+            fdict[dsname].extend([args.xrd + f for f in flist if len(f) > 1])
+
+    # pprint.pprint(fdict, depth=1)
+    return fdict
+
+
+def getFilesFromPath(args, lim=None):
+
+    fdict = {}
+    fset = []
+    with open(args.input) as fp:
+        lines = fp.readlines()
+        for line in lines:
+            if line.startswith("#") or line.strip() == "":
+                continue
+            if line.startswith("/"):
+                print(
+                    "You are trying to read files from path, but providing a dataset in DAS:\n",
+                    line,
+                )
+                print("That's not gonna work, so we exit here")
+                sys.exit(1)
+            ds = line.strip().split()
+            print("ds=", ds)
+            dataset = ds[0]
+            fdict[ds[0]] = getRootFilesFromPath(ds[1])
+
+    return fdict
+
+
+def getRootFilesFromPath(d, lim=None):
+
+    import subprocess
+
+    if "xrootd" in d:
+        sp = d.split("/")
+        siteIP = "/".join(sp[0:4])
+        pathToFiles = "/".join(sp[3:]) + "/"
+        allfiles = str(
+            subprocess.check_output(["xrdfs", siteIP, "ls", pathToFiles]), "utf-8"
+        ).split("\n")
+        # rootfiles = [siteIP+'/'+f for i,f in enumerate(allfiles) if f.endswith(".root") and (lim==None or i<lim)]
+    else:
+        siteIP = ""
+        pathToFiles = d
+        allfiles = [
+            path.join(d, f) for i, f in enumerate(listdir(d)) if f.endswith(".root")
+        ]
+
+    # print(siteIP, pathToFiles)
+    rootfiles = []
+    for file_or_dir in allfiles:
+        # print(file_or_dir)
+        if file_or_dir == "" or file_or_dir == pathToFiles:
+            continue
+        file_or_dir = siteIP + file_or_dir
+        if file_or_dir.endswith(".root"):
+            if lim == None or len(rootfiles) < lim:
+                rootfiles.append(file_or_dir)
+
+        elif not "log" in file_or_dir and not file_or_dir[-1] == "/":
+            file_or_dir = file_or_dir + "/"
+            print("file or dir:", file_or_dir)
+            if lim == None:
+                rootfiles.extend(getRootFilesFromPath(file_or_dir))
+            elif len(rootfiles) < lim:
+                rootfiles.extend(
+                    getRootFilesFromPath(file_or_dir, lim - len(rootfiles))
+                )
+
+    # print("Input path:", d)
+    # print("List of root files to be processed:\n",rootfiles)
+
+    return rootfiles
+
+
+def main(args):
+
+    if args.from_path:
+        print("do it from path: ")
+
+        fdict = getFilesFromPath(args)
+
+    else:
+
+        fdict = getFilesFromDas(args)
+
+    # print(fdict)
+    output_file = "./metadata/%s" % (args.output)
```

Comment:
path to `metadata/`  probably not necessary? it can be also stored under user directory not in metadata

Reply:
Well, in the current script it also stores output to metadata. I agree, maybe this is not necessary, but I did not want to change the behavior in this commit.